packages/clerk-js/src/utils/url.ts

```diff
@@ -369,3 +388,16 @@ export function createAllowedRedirectOrigins(
 
   return origins;
 }
+
+export const isCrossOrigin = (url: string | URL, origin: string | URL = window.location.origin): boolean => {
+  try {
+    if (isRelativeUrl(url)) {
+      return false;
+    }
+    const urlOrigin = new URL(url).origin;
+    const originOrigin = new URL(origin).origin;
```

Comment:
Shouldn't this be always `window.location.origin` ? why do we need to grab it as param ?

Reply:
To allow for easier testing without having to mock window.location :)